docs/ADR-4-Support-only-for-mainnet-and-upcoming-eras.md

```diff
@@ -86,272 +94,67 @@ makeUnsignedTx era bc = ...
 
 
 testBabbageUnsignedTx :: Either UnsignedTxError (UnsignedTx BabbageEra)
-testBabbageUnsignedTx = makeUnsignedTx CurrentEra $ defaultTxBodyContent ShelleyBasedEraBabbage
+testBabbageUnsignedTx = makeUnsignedTx BabbageEra $ defaultTxBodyContent ShelleyBasedEraBabbage
 
 testConwayUnsignedTx :: Either UnsignedTxError (UnsignedTx ConwayEra)
-testConwayUnsignedTx = makeUnsignedTx UpcomingEra $ defaultTxBodyContent ShelleyBasedEraConway
+testConwayUnsignedTx = makeUnsignedTx ConwayEra $ defaultTxBodyContent ShelleyBasedEraConway
 ```
 
-In the current state of affairs both `testBabbageUnsignedTx` and `testConwayUnsignedTx` type check without issue. However let's update the types to show how a deprecation (or a successful hardfork to the upcoming era) will show up in a user's code.
+In the current state of affairs both `testBabbageUnsignedTx` and `testConwayUnsignedTx` type check without issue.
+However let's update the types to show how a deprecation (or a successful hardfork to the upcoming era) will show up in a user's code.
 
 Changes:
-
 ```haskell
+-- this deprecation pragma deprecates both the type `BabbageEra` and the `BabbageEra` constructor
 {-# DEPRECATED BabbageEra "We are currently in the Conway era. Please update your type Signature to ConwayEra" #-}
 data BabbageEra
 ...
 ...
 data Era era where
+  -- | Previous era, do not use.
+  BabbageEra :: Era BabbageEra
   -- | The era currently active on Cardano's mainnet.
-  CurrentEra ::  Era ConwayEra
-  -- | The era planned for the next hardfork on Cardano's mainnet.
-  UpcomingEra :: Era (UninhabitableType EraCurrentlyNonExistent)
-```
-Type errors:
-
-```haskell
-In the use of type constructor or class ‘BabbageEra’
-(imported from Cardano.Api.Experimental.Eras, but defined in Cardano.Api.Protocol.AvailableEras):
-Deprecated: "We are currently in the Conway era. Please update your type Signature to ConwayEra"
-
-and
-
-internal/Cardano/Api/Experimental/Tx.hs:34:39: error: [GHC-64725]
-    • There is currently no planned upcoming era. Use CurrentEra instead.
-    • In the first argument of ‘makeUnsignedTx’, namely ‘UpcomingEra’
-      In the first argument of ‘($)’, namely ‘makeUnsignedTx UpcomingEra’
-      In the expression:
-        makeUnsignedTx UpcomingEra
-          $ defaultTxBodyContent ShelleyBasedEraConway
-   |
-34 | testConwayUnsignedTx = makeUnsignedTx UpcomingEra $ defaultTxBodyContent ShelleyBasedEraConway
-```
-
-This deprecation will happen only after the hardfork has been successful. There should be no need for consumers of the api to need the prior era's specific functionality. If users for some reason want this, we can direct them to use the ledger's api directly.
-
-## Downsides
```

Comment:
I don't think this description of downsides is relevant anymore, so I've removed it.